lib/modules/login/login_presenter.dart

```diff
@@ -22,4 +34,38 @@ class LoginPresenterImpl extends LoginPresenter
 
   @override
   final alertDialogDidClose = BehaviorSubject<void>();
+
+  @override
+  final didLogin = BehaviorSubject<void>();
+
+  @override
+  final didFailToLogin = BehaviorSubject<Exception>();
+
+  void _didInitState() {
+    view.setLoginButton(isEnabled: false);
+    view.beginAnimation();
+  }
+
+  void _didTapForgotButton() {
+    router.pushToForgotPasswordScreen(view.context);
+  }
+
+  void _didTapLoginButton(List<String> data) {
```

Comment:
a minor trick: you can name this arg as `emailAndPassword`, so when accessing, it's easier to understand than `data[0]` <- you don't know what is this field stands for 😄 
the other option is to use `Pair`, it's similar to List, but it has only 2 values, so usually, it matches this case of email + password, & easier to manage the field accessibility. This is an optional suggestion.